src/solvers/dgsem_t8code/containers.jl

```diff
@@ -0,0 +1,329 @@
+  u = unsafe_wrap(Array, pointer(_u),
+    (2, nvariables(equations), ntuple(_ -> nnodes(basis), NDIMS-1)..., n_interfaces))
+
+  _neighbor_ids = Vector{Int}(undef, 2 * n_interfaces)
+  neighbor_ids = unsafe_wrap(Array, pointer(_neighbor_ids), (2, n_interfaces))
+
+  _node_indices = Vector{NTuple{NDIMS, Symbol}}(undef, NDIMS * n_interfaces)
+  node_indices = unsafe_wrap(Array, pointer(_node_indices), (NDIMS, n_interfaces))
+
+  interfaces = T8codeInterfaceContainer{NDIMS, uEltype, NDIMS+2}(u, neighbor_ids, node_indices,
+                                                                _u, _neighbor_ids, _node_indices)
+
+  return interfaces
+end
+
+# Create mortar container and initialize mortar data.
+function init_mortars(mesh::T8codeMesh, equations, basis, elements)
+  NDIMS = ndims(elements)
+  uEltype = eltype(elements)
+
+  n_mortars = nmortars(mesh)
+
+  _u = Vector{uEltype}(undef,
+    2 * nvariables(equations) * 2^(NDIMS-1) * nnodes(basis)^(NDIMS-1) * n_mortars)
+  u = unsafe_wrap(Array, pointer(_u),
+    (2, nvariables(equations), 2^(NDIMS-1), ntuple(_ -> nnodes(basis), NDIMS-1)..., n_mortars))
+
+  _neighbor_ids = Vector{Int}(undef, (2^(NDIMS-1) + 1) * n_mortars)
+  neighbor_ids = unsafe_wrap(Array, pointer(_neighbor_ids), (2^(NDIMS-1) + 1, n_mortars))
+
+  _node_indices = Vector{NTuple{NDIMS, Symbol}}(undef, NDIMS * n_mortars)
+  node_indices = unsafe_wrap(Array, pointer(_node_indices), (NDIMS, n_mortars))
+
+  mortars = T8codeMortarContainer{NDIMS, uEltype, NDIMS+1, NDIMS+3}(u, neighbor_ids, node_indices,
+                                                                   _u, _neighbor_ids, _node_indices)
+
+  return mortars
+end
+
+# Create interface container and initialize interface data in `elements`.
+function init_boundaries(mesh::T8codeMesh, equations, basis, elements)
+  NDIMS = ndims(elements)
+  uEltype = eltype(elements)
+
+  # Initialize container.
+  n_boundaries = nboundaries(mesh)
+
+  _u = Vector{uEltype}(undef, nvariables(equations) * nnodes(basis)^(NDIMS-1) * n_boundaries)
+  u = unsafe_wrap(Array, pointer(_u),
+    (nvariables(equations), ntuple(_ -> nnodes(basis), NDIMS-1)..., n_boundaries))
+
+  neighbor_ids = Vector{Int}(undef, n_boundaries)
+  node_indices = Vector{NTuple{NDIMS, Symbol}}(undef, n_boundaries)
+  names        = Vector{Symbol}(undef, n_boundaries)
+
+  boundaries = T8codeBoundaryContainer{NDIMS, uEltype, NDIMS+1}(u, neighbor_ids,
+                                                               node_indices, names, _u)
+
+  return boundaries
+end
+
+# ============================================================================ #
+# ============================================================================ #
+
+function count_required_surfaces(mesh::T8codeMesh)
+
+  counts = trixi_t8_count_interfaces(mesh.forest)
+
+  mesh.nmortars    = counts.mortars
+  mesh.ninterfaces = counts.interfaces
+  mesh.nboundaries = counts.boundaries
+
+  return counts
+end
```

Comment:
If you change the argument, you should add a `!` to the function name.

Reply:
Good point! Updated.